pkg/runtime/interfaces.go

```diff
@@ -80,13 +80,81 @@ type ParameterCodec interface {
+}
+
+// StreamSerializerInfo contains information about a specific stream serialization format
+type StreamSerializerInfo struct {
+	SerializerInfo
+	// Framer is the factory for retrieving streams that separate objects on the wire
+	Framer
+	// Embedded is the type of the nested serialization that should be used.
+	Embedded SerializerInfo
+}
+
 // NegotiatedSerializer is an interface used for obtaining encoders, decoders, and serializers
-// for multiple supported media types.
+// for multiple supported media types. This would commonly be accepted by a server component
+// that performs HTTP content negotiation to accept multiple formats.
 type NegotiatedSerializer interface {
+	// SupportedMediaTypes is the media types supported for reading and writing single objects.
 	SupportedMediaTypes() []string
-	SerializerForMediaType(mediaType string, options map[string]string) (Serializer, bool)
-	EncoderForVersion(serializer Serializer, gv unversioned.GroupVersion) Encoder
-	DecoderToVersion(serializer Serializer, gv unversioned.GroupVersion) Decoder
+	// SerializerForMediaType returns a serializer for the provided media type. params is the set of
+	// parameters applied to the media type that may modify the resulting output. ok will be false
+	// if no serializer matched the media type.
+	SerializerForMediaType(mediaType string, params map[string]string) (s SerializerInfo, ok bool)
+
+	// SupportedStreamingMediaTypes returns the media types of the supported streaming serializers.
+	// Streaming serializers control how multiple objects are written to a stream output.
+	SupportedStreamingMediaTypes() []string
+	// StreamingSerializerForMediaType returns a serializer for the provided media type that supports
+	// reading and writing multiple objects to a stream. It returns a framer and serializer, or an
+	// error if no such serializer can be created. Params is the set of parameters applied to the
+	// media type that may modify the resulting output. ok will be false if no serializer matched
+	// the media type.
+	StreamingSerializerForMediaType(mediaType string, params map[string]string) (s StreamSerializerInfo, ok bool)
+
+	// EncoderForVersion returns an encoder that ensures objects being written to the provided
+	// serializer are in the provided group version.
+	// TODO: take multiple group versions
+	EncoderForVersion(serializer Encoder, gv unversioned.GroupVersion) Encoder
+	// DecoderForVersion returns a decoder that ensures objects being read by the provided
+	// serializer are in the provided group version by default.
+	// TODO: take multiple group versions
+	DecoderToVersion(serializer Decoder, gv unversioned.GroupVersion) Decoder
+}
+
+// StorageSerializer is an interface used for obtaining encoders, decoders, and serializers
+// that can read and write data at rest. This would commonly be used by client tools that must
+// read files, or server side storage interfaces that persist restful objects.
+type StorageSerializer interface {
```

Comment:
Probably at some point, did not expose it yet until we need it.

On Mon, Apr 25, 2016 at 11:26 AM, David Eads <notifications@github.com>
wrote:

> In pkg/runtime/interfaces.go
> <https://github.com/kubernetes/kubernetes/pull/24710#discussion_r60932967>
> :
>
> > +	StreamingSerializerForMediaType(mediaType string, params map[string]string) (s StreamSerializerInfo, ok bool)
> > +
> > +	// EncoderForVersion returns an encoder that ensures objects being written to the provided
> > +	// serializer are in the provided group version.
> > +	// TODO: take multiple group versions
> > +	EncoderForVersion(serializer Encoder, gv unversioned.GroupVersion) Encoder
> > +	// DecoderForVersion returns a decoder that ensures objects being read by the provided
> > +	// serializer are in the provided group version by default.
> > +	// TODO: take multiple group versions
> > +	DecoderToVersion(serializer Decoder, gv unversioned.GroupVersion) Decoder
> > +}
> > +
> > +// StorageSerializer is an interface used for obtaining encoders, decoders, and serializers
> > +// that can read and write data at rest. This would commonly be used by client tools that must
> > +// read files, or server side storage interfaces that persist restful objects.
> > +type StorageSerializer interface {
>
> Seems like this should have SupportedMediaTypes too.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24710/files/e4a1f3c4b34ba5ef0c8f08009e202d0b048498ff#r60932967>
>